actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -510,6 +516,24 @@ private void updateUserOrganisations(Request actorMessage) {
     }
   }
 
+  private void updateUserSelfDeclaredData(Request actorMessage, Map org, String userId) {
+    List<Map<String, Object>> declredDetails =
+        userSelfDeclarationDao.getUserSelfDeclaredFields(userId, actorMessage.getRequestContext());
+    if (!CollectionUtils.isEmpty(declredDetails)) {
+      UserDeclareEntity userDeclareEntity =
+          mapper.convertValue(declredDetails, UserDeclareEntity.class);
+      Map declaredInfo = userDeclareEntity.getUserInfo();
+      if (StringUtils.isNotEmpty(
+              (CharSequence) declaredInfo.get(JsonKey.DECLARED_SCHOOL_UDISE_CODE))
+          && !org.get(JsonKey.ID).equals(declaredInfo.get(JsonKey.DECLARED_SCHOOL_UDISE_CODE))) {
+        declaredInfo.put(JsonKey.DECLARED_SCHOOL_UDISE_CODE, org.get(JsonKey.ID));
+        declaredInfo.put(JsonKey.DECLARED_SCHOOL_NAME, org.get(JsonKey.ORG_NAME));
```

Comment:
External id should be taken instead of org id

Reply:
Done